crates/codegen/testing/src/cst_output.rs

```diff
@@ -63,6 +63,9 @@ fn collect_parser_tests(data_dir: &Path) -> Result<BTreeMap<String, BTreeSet<Str
                     .or_default()
                     .insert(test_name.to_owned());
             }
+            [.., ".gitattributes"] => {
```

Comment:
Thoughts on using unit tests for this kind of thing (like the existing `crates/solidity/outputs/cargo/tests/src/trivia.rs`) instead of using `.gitattributes`?

1. The used escape sequences are not visible/readable in the snapshot `input.sol` file. A unit test would have the explicit Rust string with escapes (`let source = "foo \\\r\n";`).
2. `.gitattributes` solves the issue for `git add`, but I worry about other tools ignoring it, like the prettier formatter, or VS Code on-save settings.

Reply:
I'm on the fence; I can understand that `trivia.rs` is more about unit testing the end of line token handling, but this is supposed to test an entire rule and fits our pre-existing snapshots rather well; we also can see that there's `\r\n` in the CST tree output and any differences will be caught during CI pipeline, also I don't expect to format or re-save this file anytime soon, as these tests are supposed to capture the input quite faithfully as we do everywhere else.

If we reverse this and if I'd committed a unit test with inline source, `parse(RuleKind::StringLiteral)` that then asserts the tree shape and verifies that the newline and escape are properly captured and handled, I believe we'd think "this should've been a snapshot test", so I'm inclined to leave it the way it is but I'm open.